go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -659,75 +682,132 @@ func getPolicyType(policy *knet.NetworkPolicy) (bool, bool) {
 
 // getNewLocalPolicyPorts will find and return port info for every given pod obj, that is not found in
 // np.localPods.
-// if there are problems with fetching port info from logicalPortCache, pod will be added to errObjs.
-func (oc *DefaultNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
-	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errObjs []interface{}) {
+// if there are problems with fetching port info from logicalPortCache, error will be added to returned error array.
+func (bnc *BaseNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
+	objs ...interface{}) (map[string]string, []string, []error) {
+	var errs []error
 
 	klog.Infof("Processing NetworkPolicy %s/%s to have %d local pods...", np.namespace, np.name, len(objs))
-	policyPortUUIDs = make([]string, 0, len(objs))
-	policyPortsToUUIDs = map[string]string{}
+	policyPortUUIDs := []string{}
+	policyPortsToUUIDs := map[string]string{}
 
 	for _, obj := range objs {
 		pod := obj.(*kapi.Pod)
-
-		logicalPortName := util.GetLogicalPortName(pod.Namespace, pod.Name)
-		if _, ok := np.localPods.Load(logicalPortName); ok {
-			// port is already added for this policy
-			continue
-		}
-
 		if pod.Spec.NodeName == "" {
 			// pod is not yet scheduled, will receive update event for it
 			continue
 		}
 
 		// Skip pods that will never be present in logicalPortCache,
 		// e.g. hostNetwork pods, overlay node pods, or completed pods
-		if !oc.podExpectedInLogicalCache(pod) {
+		if !bnc.podExpectedInLogicalCache(pod) {
 			continue
 		}
 
-		// Add pod to errObjs for retry if
-		// 1. getting pod LSP from the cache fails,
-		// 2. the gotten LSP is scheduled for removal (stateful-sets).
-		portInfo, err := oc.logicalPortCache.get(pod, types.DefaultNetworkName)
-		if err != nil {
-			klog.Warningf("Failed to get LSP for pod %s/%s for networkPolicy %s, err: %v",
-				pod.Namespace, pod.Name, np.name, err)
-			errObjs = append(errObjs, pod)
+		if !bnc.IsSecondary() {
+			portInfo, err := bnc.getNewLocalPolicyPortsForNAD(np, pod, types.DefaultNetworkName)
+			if err != nil {
+				errs = append(errs, err)
+			} else if portInfo != nil {
+				policyPortUUIDs = append(policyPortUUIDs, portInfo.uuid)
+				policyPortsToUUIDs[portInfo.name] = portInfo.uuid
+			}
 			continue
 		}
 
-		// Add pod to errObjs if LSP is scheduled for deletion
-		if !portInfo.expires.IsZero() {
-			klog.Warningf("Stale LSP %s for network policy %s found in cache",
-				portInfo.name, np.name)
-			errObjs = append(errObjs, pod)
+		// secondary networks
+		on, networkMap, err := util.GetPodNADToNetworkMapping(pod, bnc.NetInfo)
+		if err != nil || !on {
```

Comment:
Why are we not returning the error here. Add a comment with an explanation please.